src/main/java/com/notably/model/block/BlockImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.notably.model.block;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents a Block in the Notably data structure.
+ */
+public class BlockImpl implements Block {
+    private List<Block> children;
+    private Title title;
+    private Body body;
+    private Block parent;
+
+    /**
+     * Initializes a default block.
+     * Used when creating a root block and creating a block without the optional body argument.
+     */
+    public BlockImpl(Block parent, Title title) {
+        this.title = title;
+        this.parent = parent;
+    }
+
+    /**
+     * Initializes a block with body content.
+     * Used when creating a block with additional body argument.
+     */
+    public BlockImpl(Block parent, Title title, Body body) {
+        this.title = title;
+        this.parent = parent;
+        this.body = body;
+    }
+
+    @Override
+    public Title getTitle() {
+        return this.title;
+    }
+
+    @Override
+    public Body getBody() {
+        return this.body;
+    }
+
+    @Override
+    public Optional<Block> getParent() {
```

Comment:
btw Is there a chance parent can be null? If so, `Optional.of(null)` would result in an error as the argument needs to be non-null. If this is the case, you can consider returning `Optional.ofNullable(parent)` or return an `Optional.empty(`).

Reply:
Currently the only chance a parent is null is if the currently open block is a `root` block. 

However, you're right about the NPE error! I intended to use the `Optional.empty()` method during navigation as well. I'll get it corrected! Thanks for pointing it out

Reply:
@HemanshuGandhi Corrected!